pkg/upgrade/upgrade.go

```diff
@@ -63,10 +63,19 @@ func ManagedInstances(ctx context.Context, c client.Client, reader client.Reader
 			// nothing to do at this level, just go to the next instance
 			continue
 		}
-
 		if !reflect.DeepEqual(jaeger, j) {
+			version := jaeger.Status.Version
 			// the CR has changed, store it!
-			if err := c.Update(ctx, &jaeger); err != nil {
+			if err := c.Update(ctx, &jaeger); err == nil {
+				jaeger.Status.Version = version
+				if err := c.Status().Update(ctx, &jaeger); err != nil {
```

Comment:
s/Update/Patch/ ?

Reply:
I can do the change but What would be the benefit of doing patch for this case?

Reply:
Nevermind, I did the change.